### DISCUSSION TITLE
Github repo per branch in edit_uri field

### DISCUSSION THREAD

**Discussion by jasonnovichRunAI** [squidfunk/mkdocs-material#5292] [Q&A]
Hello,
Not sure if this question was asked, if it was, please point me to the right discussion. If not, here goes...

I am deploying MKDocs with versioning using mike based on branches in my Github repo.
For each version of the docs I create a version branch usually titled vX.YY - this is to make sure that if there are any changes to the product that need to be backdated to previous versions, I can make the changes and redeploy. For deployment I have a Github action that deploys each set of docs using mike (pick the branch and build the docs from that branch). The final deployment is done using gh-pages to a custom domain. Each branch build is deployed that way and the selector button in the menu works.

I enabled the `edit_uri` for the deployment, and currently it points to `master` (sorry old repo naming). If I press the edit button on a new page in the docs I get a 404 error, which is expected becaue the new pages have not been merged into `master`. 

Is it possible to have the `edit_uri` point to the specific version branch where the edit button was pressed? This is important as our customers, or other stakeholders can request changes in the docs that would apply to the version of the docs they are reading. The company is planning on keeping up to 8 versions (latest -7). 

Thanks,
Jason

**Comment by squidfunk** [comment]
You can configure [`edit_uri`](https://www.mkdocs.org/user-guide/configuration/#edit_uri) in `mkdocs.yml` prior to deployment and change it to the branch. `master` is just the default, e.g.:

``` yaml
edit_uri: blob/vX.YY/docs/
```

**Reply by jasonnovichRunAI** [reply]
Thanks!  i was hoping to use some kind of variable that would allow me to configure it so that I do not have to manually configure it for each branch, and then when I merge into master "remember" to change it there.

**Reply by squidfunk** [reply]
MkDocs supports the use of [environment variables](https://www.mkdocs.org/user-guide/configuration/#environment-variables), so you could effectively use something like [`GITHUB_REF`](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) when building from a GitHub Action.

**Reply by jasonnovichRunAI** [reply]
I will certainly give that a try, and let you know how that worked.
Thanks!  8-)
